Formula/balena-cli.rb

```diff
@@ -18,7 +18,7 @@ class BalenaCli < Formula
     sha256 "0e670f3cc482565ddc1a166316407fab2fd008f12128a06f7f73c3bcf07a641d" => :high_sierra
   end
 
-  depends_on "node"
+  depends_on "node@12"
```

Comment:
Nice 👍  I am one of the maintainers of https://github.com/balena-io/balena-cli and I had been meaning for a while to ask you guys to pin the version of Node. Support for Node v14 is coming soon (balena-io/balena-cli/issues/2079), but not just yet. :-)

Reply:
We don't pin the dependencies unless a build stops working with a certain dependency.

Reply:
It makes sense. By the way, starting with `balena-cli` version 12.38.4, we've restricted the `engines` value in `package.json`:
> https://github.com/balena-io/balena-cli/blob/v12.38.4/package.json#L86-L88
> `"engines": { "node": ">=10.20.0 <13.0.0" },`

Meaning that, if `balena-cli` is executed with Node.js v13 or v14 or v15 or later, it now prints a warning:
```
$ balena version -a
------------------------------------------------------------------------------
Warning: Node version "v14.15.4" does not match required versions ">=10.20.0 <13.0.0".
This may cause unexpected behavior. To upgrade Node, visit:
https://nodejs.org/en/download/
------------------------------------------------------------------------------
balena-cli version "12.38.3"
Node.js version "14.15.4"
```
(I understand that the warning message is the default printed by Node.js, and may vary a bit depending on the Node.js interpreter.)

Of course we will amend it again as soon a support for Node v14 is added.

For anyone who is interested, I would suggest rewriting the `test` section of the `balena-cli.rb` formula so that it is made to fail if the word "warning" is found in the command output.